### ISSUE TITLE
New session ID on every page load

### ISSUE THREAD

**Issue by darkalchemy** [middlewares/php-session#13]
https://github.com/middlewares/php-session/blob/d95c5d321ba0b367d181391df5d85e3e82fe05d7/src/PhpSession.php#L149-L151

If the session id isn't stored in the cookie, then a new session id is generated on every request. So it's like running regenerate id on every request. I think this should be set to true, not false. Or, maybe removed to allow the choice if we want to store it or not.

Thanks

**Comment by oscarotero** [comment]
The session id is stored in cookies but not automatically by php but by an internal function:

https://github.com/middlewares/php-session/blob/d95c5d321ba0b367d181391df5d85e3e82fe05d7/src/PhpSession.php#L118-L127

Is this code is executed in your use case?

**Comment by darkalchemy** [comment]
Yes, it's executed. It's being set as a cookie value, but not by php. So, I don't think php is looking for the cookie as it didn't set it. I am not 100% sure on that. But, I have tested clean code that uses this middleware and a new session id is generated on every request. 

Thanks

**Comment by oscarotero** [comment]
The session is recovered here:
https://github.com/middlewares/php-session/blob/d95c5d321ba0b367d181391df5d85e3e82fe05d7/src/PhpSession.php#L93-L101

The middleware reads the session name and id from the cookie before starting the session, so it could be restored. Note also the option to regenenerate the id in a certain interval.

**Comment by darkalchemy** [comment]
I'm not able to test it now, but where's it getting the name from? It may be empty and therefore not able to get the I'd from the cookie?

I'll check this later today. 

Thanks again

**Comment by oscarotero** [comment]
You can set a specific name or get the default `session_name()` as fallback.

**Comment by oscarotero** [comment]
Ok, if you think this can be improved, feel free to send a PR.
Thanks!
